### ISSUE TITLE
Qodot 4 Errors (and some fixes)

### ISSUE THREAD

**Issue by IAmTraffic** [QodotPlugin/qodotplugin.github.io#7]
First time using Godot 4, installed Qodot via the AssetLib. Installation went fine, but when I tried enabling the plugin I got the following error:

` res://addons/qodot/src/qodot_plugin.gd:16 - Parse Error: The function signature doesn't match the parent. Parent signature is "_handles(Object) -> bool".
  res://addons/qodot/src/qodot_plugin.gd:19 - Parse Error: The function signature doesn't match the parent. Parent signature is "_edit(Object) -> void".`
  
Changing the parameters on lines 16 and 19 of qodot_plugin.gd from `object: Variant` to `object` seems to have fixed the issue for me.

Additionally, I was unable to edit qodot_trenchbroom_config_folder.tres within the editor. Opening the file in an outside text editor and changing line 4 to:
`[ext_resource type="Texture" path="res://icon.svg" id="2"]`
did the trick for me, but you could also just change the path to reference the Qodot icon.png.

I also kept getting `No loader found for resource` errors for .cs files, which I wasn't able to solve. It looks like the code all follows the [documentation](https://docs.godotengine.org/en/latest/tutorials/scripting/cross_language_scripting.html), so I think it may just be an issue with Godot 4? I suppose time will tell.

Hope to be able to use this plugin in Godot 4 soon!

**Comment by DeerTears** [comment]
Thanks for reporting those issues! The first ones on enable should have been fixed by #15, I recently just edited the config_folder tool to use the new icon.svg instead of .png.

`No loader found for resource` I think either means you still need to create a C# solution for the project (Project > C# > Create C# Solution), or that Godot won't run certain functions if the plugin itself has errors still.

These aren't necessarily documentation issues, if you run into more issues like these feel free to file them here: https://github.com/QodotPlugin/Qodot
